File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.py
class Solution(object):
    def Search(self,tar,arr,i):
        l = 0
        r = len(arr) - 1
        while(l <= r):
            mid = (l+r)//2
            if(arr[mid] == tar and mid != i):
                return True
            elif arr[mid] < tar:
                l = mid+1
            else:
                r = mid - 1
        return False
    def checkIfExist(self, arr):
        arr.sort()
        for i in range(len(arr)):
            if(self.Search(arr[i]*2,arr,i)):
                return True
        return False
    
        

File: 1468-check-if-n-and-its-double-exist/check-if-n-and-its-double-exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        for(int i = 0;i < arr.size();i++)
        {
            if(Search(arr[i]*2,arr,i))
                return true;
        }
        return false;
    }
    bool Search(int tar,const vector<int> &arr,int k)
    {
        int l = 0,r = arr.size()-1;
        while(l <= r)
        {
            int mid = l + (r-l)/2;
            if(arr[mid] == tar && mid != k)
                return true;
            else if(arr[mid] > tar)
                r = mid - 1;
            else
                l = mid + 1;
        }
        return false;
    }
};